src/api.js

```diff
@@ -2087,6 +2087,91 @@ api.declare({
   return res.reply(result);
 });
 
+/** Get a worker-type from a provisioner */
+api.declare({
+  method:     'get',
+  route:      '/provisioners/:provisionerId/worker-types/:workerType',
+  name:       'getWorkerType',
+  stability:  API.stability.experimental,
+  output:     'workertype-response.json#',
+  title:      'Get a worker-type',
+  description: [
+    'Get a worker-type from a provisioner.',
+  ].join('\n'),
+}, async function(req, res) {
+  const {provisionerId, workerType} = req.params;
+
+  const wType = await this.WorkerType.scan({
+    provisionerId,
+    workerType,
+    expires: Entity.op.greaterThan(new Date()),
+  });
+
+  const result = wType.entries.length ? wType.entries[0].json() : {};
+
+  if (!wType.entries.length) {
+    return res.reportError('ResourceNotFound',
+      'Worker-type {{workerType}} with Provisioner {{provisionerId}} not found. Are you sure it was created?', {
+        workerType,
+        provisionerId,
+      },
+    );
+  }
+
+  return res.reply(result);
+});
+
+/** Update a worker-type */
+api.declare({
+  method:     'put',
+  route:      '/provisioners/:provisionerId/worker-types/:workerType',
+  name:       'updateWorkerType',
+  stability:  API.stability.experimental,
+  scopes:     [
+    [
+      'queue:declare-worker-type:<provisionerId>/<workerType>#<property>',
+    ],
+  ],
+  deferAuth:  true,
+  output:     'workertype-response.json#',
+  input:      'update-workertype-request.json#',
+  title:      'Update a worker-type',
+  description: [
+    'Update a worker-type from a provisioner.',
+  ].join('\n'),
+}, async function(req, res) {
+  const {provisionerId, workerType} = req.params;
+  const {stability, description, expires} = req.body;
+
+  const wType = await this.WorkerType.load({provisionerId, workerType}, true);
+
+  // Authenticate request by providing parameters
+  const requestAllowed = req.satisfies({provisionerId, workerType, property: stability}) &&
+    req.satisfies({provisionerId, workerType, property: description}) &&
+    req.satisfies({provisionerId, workerType, property: expires});
```

Comment:
These would be `property: 'stability'`, `property: 'description'`, etc.  That is, the completed scopes are things like `queue:declare-worker-type:aws-provisioner-v1/gecko-b-1-w2008#expires`.

Also, if the property isn't provided in the request, we shouldn't check for the scope.  So a request with `{description: 'This worker type is great'}` would only require `queue:declare-worker-type:aws-provisioner-v1/gecko-b-1-w2008#desscription`.  But if the request also included expires, the `#expires` scope would be required too.

Reply:
This probably should have a few sentences in the API description to explain it, too.

Reply:
That was a stupid mistake from my part. Thanks!